source/sigv4.c

```diff
@@ -1656,7 +1659,8 @@ static void generateCredentialScope( const SigV4Parameters_t * pSigV4Params,
         if( sigV4Status == SigV4Success )
         {
             /* Replacing the last ';' with '\n' as last header should not have ';'. */
-            *( canonicalRequest->pBufCur - 1 ) = '\n';
+            pBufPtr = ( char * ) ( &canonicalRequest->pBufProcessing[ canonicalRequest->pBufCur - ( char * ) ( &canonicalRequest->pBufProcessing ) - 1 ] );
+            *pBufPtr = '\n';
```

Comment:
Can this not be simplified like so?
```suggestion
            canonicalRequest->pBufCur[ -1 ] = '\n';
```

I think it is too hard to understand right now by reading it.

Reply:
Update in new commit .